library/core/src/hint.rs

```diff
@@ -163,6 +163,8 @@ pub const unsafe fn unreachable_unchecked() -> ! {
 #[inline(always)]
 #[stable(feature = "renamed_spin_loop", since = "1.49.0")]
 pub fn spin_loop() {
+    #![cfg_attr(not(bootstrap), allow(unnecessary_block))]
```

Comment:
So, this is an interesting use case - on any given `cfg`, these blocks are "unnecessary", but overall they make it much easier to write this function. An alternative would be something like `cfg-if`, but that isn't in std.